### PR TITLE
Support GEOSHAPE field type in RediSearch

### PR DESCRIPTION
**Pull request by sazzad16**

### CODE REVIEW

**Comment by sazzad16** [comment]
@chayim @uglide :bell: 

@bsbodden :eye:

File: pom.xml [hunk 1]
```diff
@@ -73,6 +73,21 @@
 			<version>2.10.1</version>
 		</dependency>
 
+		<!-- UNIX socket connection support -->
+		<dependency>
+			<groupId>com.kohlschutter.junixsocket</groupId>
+			<artifactId>junixsocket-core</artifactId>
```
Comment (chayim) [inline pom.xml:79]: If we're introducing this, why can we use the latest, 2.8.1?

File: src/test/java/redis/clients/jedis/examples/GeoShapeFieldsUsageInRediSearch.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import redis.clients.jedis.UnifiedJedis;
+import redis.clients.jedis.search.FTSearchParams;
+import redis.clients.jedis.search.SearchResult;
+import redis.clients.jedis.search.schemafields.GeoShapeField;
+
+import static java.util.Collections.singletonMap;
+import static org.junit.Assert.assertEquals;
+import static redis.clients.jedis.search.RediSearchUtil.toStringMap;
+
+/**
```
Comment (chayim) [inline src/test/java/redis/clients/jedis/examples/GeoShapeFieldsUsageInRediSearch.java:21]: /**
 * As of RediSearch 2.8.4, advanced GEO querying with GEOSHAPE fields is supported.
 *
 *Any object/library producing
 * <a href="https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry">
 * in a well-known text (WKT)</a> in {@code toString()} method can be used.
 *
 * This example uses the
 * <a href="https://github.com/locationtech/jts">JTS</a> library.
 * <pre>
 * {@code
 * <dependency>
 *   <groupId>org.locationtech.jts</groupId>
 *   <artifactId>jts-core</artifactId>
 *   <version>1.19.0</version>
 * </dependency>
 * }
 * </pre>
 */

File: src/test/java/redis/clients/jedis/examples/GeoShapeFieldsUsageInRediSearch.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ * </pre>
+ */
+public class GeoShapeFieldsUsageInRediSearch {
+
+  public static void main(String[] args) {
+
+    // We'll create geometry objects with GeometryFactory
+    final GeometryFactory factory = new GeometryFactory();
+
+    final String host = "<HOST>";
```
Comment (chayim) [inline src/test/java/redis/clients/jedis/examples/GeoShapeFieldsUsageInRediSearch.java:47]: localhost at least?

**Comment by chayim** [comment]
@uglide IMHO docs... Who can help?

File: pom.xml (hunk 1 above)
Comment (sazzad16) [reply pom.xml:79]: @chayim We're not introducing this. I have just moved it up to keep "the test dependencies that can be used as optional dependencies" together.

Comment (chayim) [reply pom.xml:79]: Diffs clearly aren't showing right here :(.  Okay, so same question - if it's a test dependency can we upgrade this to latest while we're here?

Comment (sazzad16) [reply pom.xml:79]: Done.

**Pull request merged**